pkg/agent/openflow/network_policy.go

```diff
@@ -991,6 +999,49 @@ func (c *client) InstallPolicyRuleFlows(rule *types.PolicyRule) error {
 	return nil
 }
 
+// check is the rule is egress for igmp, caller will return empty flow if it is
+func (f *featureNetworkPolicy) isIGMPEgressRule(tableID uint8, isIngress, isIGMP bool) bool {
+	// If multicast is enabled, and the rule is igmp and egress rule, this function will return true.
+	// Otherwise, it will return false.
+	if !f.enableMulticast || isIngress || !isIGMP {
+		return false
+	}
+	return true
+}
+
+// check if the rule is ingress for igmp, caller will generate conjunction flows if it is
+func (f *featureNetworkPolicy) isIGMPIngressRule(tableID uint8, isIngress, isIGMP bool) bool {
+	// If multicast is enabled, and the rule is igmp and ingress, and table is MulticastIGMPIngressTable,
+	// the rule is ingress for igmp.
+	// Otherwise, it is not.
+	if !f.enableMulticast || !isIngress || !isIGMP {
+		return false
+	}
+	return tableID == MulticastIGMPIngressTable.GetID()
+}
+
+// check if the rule is egress for multicast udp traffic, caller will generate conjunction flows if it is
```

Comment:
UDP

Reply:
But why it is only UDP? The rule does not handle IP multicast packets?

Reply:
For multicast, we use IGMP to manager the group members, and for normally multicast packets, it is on UDP. Here handles the normal multicast UDP packets.

Reply:
No, multicast can be raw IP packets, not necessarily UDP.

Reply:
thanks, will update here